src/library/dao/trackdao.cpp

```diff
@@ -1622,14 +1622,16 @@ void TrackDAO::markTrackLocationsAsDeleted(const QString& directory) {
     }
 }
 
-// Look for moved files. Look for files that have been marked as "deleted on disk"
-// and see if another "file" with the same name and filesize exists in the track_locations
-// table. That means the file has moved instead of being deleted outright, and so
-// we can salvage your existing metadata that you have in your DB (like cue points, etc.).
-void TrackDAO::detectMovedFiles(QSet<int>* pTracksMovedSetOld, QSet<int>* pTracksMovedSetNew) {
-    //This function should not start a transaction on it's own!
-    //When it's called from libraryscanner.cpp, there already is a transaction
-    //started!
+// Look for moved files. Look for files that have been marked as
+// "deleted on disk" and see if another "file" with the same name and
+// files size exists in the track_locations table. That means the file has
+// moved instead of being deleted outright, and so we can salvage your
+// existing metadata that you have in your DB (like cue points, etc.).
+bool TrackDAO::detectMovedFiles(QSet<int>* pTracksMovedSetOld,
+        QSet<int>* pTracksMovedSetNew, volatile const bool* pCancel) {
```

Comment:
It should be possible to write a MOCK `TrackInfoObject` class that we can use to fill a library in unit-tests. This should allow us to create arbitrary library states. Ideally we could have a function that takes a csv formated string like this.

```csv
Title, Location, Arists
foo, /bar/foo.mp3, blub
bar, /foo/bar.mp3, blub
foo, /foo/foo.mp3, blub
bar, /bar/bar.mp3, blub
```

This defines the state of the library accurately and we can check if these functions do what they are supposed to do. But that would be for another PR.